### PR TITLE
Model now sends didUnload event when record is removed

### PR DESCRIPTION
**Pull request by gnarf**

### CODE REVIEW

File: lib/ember-orbit/record-array-manager.js [hunk 1]
```diff
@@ -35,11 +35,11 @@ var RecordArrayManager = Ember.Object.extend({
    @private
    */
   _processChanges: function() {
-    forEach(this.changes, function(change) {
-      this._processChange(change.record, change.operation);
-    }, this);
+    var change;
```
Comment (gnarf) [inline lib/ember-orbit/record-array-manager.js]: I figured one less unnecessary closure/iterator in the "hot path" of property change might be useful, all tests still pass.  Happy to undo/do alone.

**Comment by gnarf** [comment]
I'd really like to be able to bump ember-orbit and pick this up asap, whatever you need from me let me know!

File: lib/ember-orbit/record-array-manager.js (hunk 1 above)
Comment (dgeb) [inline lib/ember-orbit/record-array-manager.js]: I would have preferred a separate commit, but not a big deal.

**Comment by dgeb** [comment]
Looks great @gnarf - thanks! New version forthcoming very soon.

**Comment by dgeb** [comment]
v0.5.1 has been tagged and pushed :shipit:

**Pull request merged**